packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -154,8 +154,32 @@ const toggleBlock = (editor, value) => {
     // When there is a selection, update the existing block in the tree
     Transforms.setNodes(editor, blockProperties);
   } else {
+    // Retrieve the last node inserted and its path
+    const [, lastNodePath] = Editor.last(editor, []);
+    const [parentNode] = Editor.parent(editor, lastNodePath, {
+      match: (node) => node.type !== 'text', // makes sure we get a block node, not an inline node
+    });
+    // Remove the last node inserted
+    Transforms.removeNodes(editor, {
+      void: true,
+      hanging: true,
+      at: {
+        anchor: { path: lastNodePath, offset: 0 },
+        focus: { path: lastNodePath, offset: 0 },
+      },
+    });
     // Otherwise, add a new block to the tree
```

Comment:
Could you explain a bit more _why_ you're doing this in the comment? (pretty much your PR title/description)

And the "otherwise" in the comment can be removed I think, it's no longer at the start of the else block so it's confusing